Ensuring deleted cookies are recognized by the browser by ensuring the MaxAge within CookieOptions is set to null.

MattyLeslie opened the pull request:
**Reset MaxAge to Null on Cookie Deletion**

**Summary:**
This PR aims to ensure reliable cookie deletion by setting MaxAge to null in the CookieOptions used by ResponseCookies.Delete, addressing inconsistencies identified in issue #52159.

**Description:**
To resolve inconsistencies in cookie deletion across browsers, this update modifies the ResponseCookies.Delete method to reset MaxAge to null. Previously, retained MaxAge values could prevent browsers from recognizing that cookies were meant to be deleted.

**Technical Changes:**

Enforced MaxAge = null in CookieOptions during cookie deletion..

Fixes
#52159 by standardizing cookie deletion behavior across all browsers.

MattyLeslie committed:
Explicitly appending the MaxAge as null at the end of the delete method to ensure deletion within the browser

MattyLeslie committed:
Ensuring the IsEssential property of CookieOptions is carried

MattyLeslie committed:
Merge branch 'dotnet:main' into #52159

MattyLeslie committed:
syntax error.

MattyLeslie committed:
Merge branch 'dotnet:main' into #52159

MattyLeslie committed:
Simplifying the change to just override MaxAge and Expires when appending

MattyLeslie committed:
Merge branch 'dotnet:main' into #52159

amcasey commented:
CI failure looks like network flakiness.  Re-running.

amcasey commented on src/Http/Http/src/Internal/ResponseCookies.cs:
```diff
@@ -163,6 +163,7 @@ public void Delete(string key, CookieOptions options)
         Append(key, string.Empty, new CookieOptions(options)
         {
             Expires = DateTimeOffset.UnixEpoch,
+            MaxAge = null,
```
```suggestion
            MaxAge = null, // Some browsers require this (https://github.com/dotnet/aspnetcore/issues/52159)
```

amcasey commented:
I like it!  Do we need to update [here](https://github.com/dotnet/aspnetcore/blob/561f135501c5cbf1bb6456d494669a35a088b493/src/Shared/ChunkingCookieManager/ChunkingCookieManager.cs#L306) and [here](https://github.com/dotnet/aspnetcore/blob/561f135501c5cbf1bb6456d494669a35a088b493/src/Shared/ChunkingCookieManager/ChunkingCookieManager.cs#L333) as well?

MattyLeslie committed:
Adding insightful comment to ResponseCookies.cs

MattyLeslie committed:
Ammending the methods DeleteCookie() and DeleteChunks() to ensure MaxAge of cookieOptions is set to null upon deletion

MattyLeslie commented:
Thanks for the review @amcasey, I believe the `ChunkingCookieManager.cs` will need to make use of the same logic when deleting chunks, I have updated the relevant areas and added the recommended comments.

amcasey reviewed (approved):
Thanks!
